kong/clustering/version_negotiation/init.lua

```diff
@@ -0,0 +1,355 @@
+
+local cjson = require "cjson.safe"
+local pl_file = require "pl.file"
+local http = require "resty.http"
+
+local constants = require "kong.constants"
+local clustering_utils = require "kong.clustering.utils"
+
+local ngx = ngx
+local ngx_log = ngx.log
+local ngx_ERR = ngx.ERR
+local ngx_DEBUG = ngx.DEBUG
+local _log_prefix = "[version-negotiation] "
+
+local KONG_VERSION
+local CLUSTERING_SYNC_STATUS = constants.CLUSTERING_SYNC_STATUS
+
+
+local _M = {}
+
+local function validate_connection_type()
```

Comment:
nit:
The code is validating a request and not a connection.
```suggestion
local function validate_request_type()
```